enumerables.rb

```diff
@@ -0,0 +1,99 @@
+# rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    length.times do |i|
+      yield to_a[i]
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    length.times do |i|
+      yield to_a[i], i
+    end
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    result = []
+    to_a.my_each { |i| result << i if yield(i) }
+    result
+  end
+
+  def my_all
+    return to_enum(:my_all) unless block_given?
+
+    result = true
+    my_each { |i| result = false unless yield(i) }
+    result
+  end
+
+  def my_any
+    return to_enum(:my_any) unless block_given?
+
+    result = false
+    my_each { |i| result = true if yield(i) }
+    result
+  end
```

Comment:
- [ ] `my_any?` is not working as the original `any?` method, also is not passing all test cases. 

### `any?` is an Enumerable that has these characteristics:
- Returns `true` if at least one element is not false.
```
[1, false, nil].any?
# true
```
- Returns `false` if all elements are false.
```
[nil, false, nil].any?
# false
```
- it doesn't change the original array/range/hash
- it can accept as arguments a Class, a RegEx, or a pattern, and returns true or false depending if the values in the array/range/hash match the condition
- it can use a block but it is not necessary to make the method work

[Enumerable#any?](https://ruby-doc.org/core-2.7.2/Enumerable.html#method-i-any-3F)